Content.Client/CartridgeLoader/Cartridges/CriminalRecordsCartridgeUiFragment.xaml.cs

```diff
@@ -0,0 +1,148 @@
+    }
+
+
+    public void UpdateState(CriminalRecordsCartridgeUiState state)
+    {
+        foreach (var (stationRecord, criminalRecord) in state.Wanted)
+        {
+            AddWanted(stationRecord, criminalRecord);
+        }
+
+        foreach (var (stationRecord, criminalRecord) in state.Detained)
+        {
+            AddDetained(stationRecord, criminalRecord);
+        }
+    }
+
+    private void AddWanted(GeneralStationRecord stationRecord, CriminalRecord criminalRecord)
+    {
+        var nameLabel = new Label()
+        {
+            Text = stationRecord.Name,
+            HorizontalExpand = true,
+            ClipText = true,
+        };
+
+        Wanted.AddChild(nameLabel);
+
+        var jobContainer = new BoxContainer()
+        {
+            Orientation = LayoutOrientation.Horizontal,
+            HorizontalExpand = true,
+        };
+
+        Wanted.AddChild(jobContainer);
+
+        var jobLabel = new Label()
+        {
+            Text = stationRecord.JobTitle,
+            HorizontalExpand = true,
+            ClipText = true,
+        };
+
+
+        if (!_prototypeManager.TryIndex<StatusIconPrototype>(
+              stationRecord.JobIcon,
+              out var proto
+              ))
+        {
+            jobContainer.AddChild(jobLabel);
+            return;
+        }
+
+        var jobIcon = new TextureRect()
+        {
+            TextureScale = new Vector2(2f, 2f),
+            VerticalAlignment = VAlignment.Center,
+            Texture = _spriteSystem.Frame0(proto.Icon),
+            Margin = new Thickness(5, 0, 5, 0),
+        };
+
+        jobContainer.AddChild(jobIcon);
+        jobContainer.AddChild(jobLabel);
+    }
+
+    private void AddDetained(GeneralStationRecord stationRecord, CriminalRecord criminalRecord)
+    {
+        var nameLabel = new Label()
+        {
+            Text = stationRecord.Name,
+            HorizontalExpand = true,
+            ClipText = true,
+        };
+
+        Detained.AddChild(nameLabel);
+
+        var jobContainer = new BoxContainer()
+        {
+            Orientation = LayoutOrientation.Horizontal,
+            HorizontalExpand = true,
+        };
+
+        Detained.AddChild(jobContainer);
+
+        var jobLabel = new Label()
+        {
+            Text = stationRecord.JobTitle,
+            HorizontalExpand = true,
+            ClipText = true,
+        };
+
+
+        if (!_prototypeManager.TryIndex<StatusIconPrototype>(
+              stationRecord.JobIcon,
+              out var proto
+              ))
+        {
+            jobContainer.AddChild(jobLabel);
+            return;
+        }
+
+        var jobIcon = new TextureRect()
+        {
+            TextureScale = new Vector2(2f, 2f),
+            VerticalAlignment = VAlignment.Center,
+            Texture = _spriteSystem.Frame0(proto.Icon),
+            Margin = new Thickness(5, 0, 5, 0),
+        };
+
+        jobContainer.AddChild(jobIcon);
+        jobContainer.AddChild(jobLabel);
+    }
+
```

Comment:
Is there a way to pass the Wanted & Detained control as a function parameter?

Reply:
Use the List typing that you made them out of?

Reply:
> Use the List typing that you made them out of?

Thank you!